### PR TITLE
Insertsort

### PR DESCRIPTION
**Pull request by varchenkomichael**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+'.idea' 
+
+'git push --set-upstream origin insertsort'
```
Comment (RubikOM) [inline .gitignore:3]: What does this line do?

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,5 @@
+'.idea' 
+
+'git push --set-upstream origin insertsort' 
+'sortproject.xml'
```
Comment (RubikOM) [inline .gitignore]: What is this file and why does it need to be here?

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,5 @@
+'.idea' 
+
+'git push --set-upstream origin insertsort' 
+'sortproject.xml' 
+'.idea','target','sortproject.iml'
```
Comment (RubikOM) [inline .gitignore]: You could go for something like `*.iml`
`.idea` is duplicated at line 1

File: src/test/java/com/varchenko/InsertSortTest.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+import org.junit.jupiter.api.Test;
+
+import java.lang.reflect.Array;
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+
+class InsertSortTest {
+    private final InsertSort insertSort = new InsertSort();
```
Comment (RubikOM) [inline src/test/java/com/varchenko/InsertSortTest.java:10]: Please add tests for `null` and for passing empty array

**Review by RubikOM — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (varchenkomichael) [inline .gitignore]: done

File: .gitignore (hunk 2 above)
Comment (varchenkomichael) [reply .gitignore]: done

File: .gitignore (hunk 3 above)
Comment (varchenkomichael) [reply .gitignore]: done

File: src/test/java/com/varchenko/InsertSortTest.java (hunk 4 above)
Comment (varchenkomichael) [reply src/test/java/com/varchenko/InsertSortTest.java:10]: done

File: src/test/java/com/varchenko/InsertSort.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package com.varchenko;
+
+import java.util.Arrays;
+
+public class InsertSort {
+    public int [] arr (int [] arr) {
```
Comment (RubikOM) [inline src/test/java/com/varchenko/InsertSort.java]: Why is this method called arr? :D

**Review by RubikOM — CHANGES_REQUESTED** [review]

File: src/test/java/com/varchenko/InsertSort.java (hunk 5 above)
Comment (varchenkomichael) [reply src/test/java/com/varchenko/InsertSort.java]: done

File: src/test/java/com/varchenko/FibonacciSequenceTest.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+import org.junit.jupiter.api.Test;
+import static com.varchenko.FibonacciSequence.fib;
+
+public class FibonacciSequenceTest {
+    FibonacciSequence fibonacciSequence = new FibonacciSequence() ;
+        @Test
+        void Fibonacci_numbersFrom0to10() {
+            for (int index = 0; index < 10; index++) {
+
+                int expectedResult = fib(index);
```
Comment (RubikOM) [inline src/test/java/com/varchenko/FibonacciSequenceTest.java]: You have to set the expected value by yourself.
Right now, you just test that your method returns the same value each time.

File: src/test/java/com/varchenko/FibonacciSequenceTest.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+import static org.junit.jupiter.api.Assertions.*;
+import org.junit.jupiter.api.Test;
+import static com.varchenko.FibonacciSequence.fib;
+
+public class FibonacciSequenceTest {
+    FibonacciSequence fibonacciSequence = new FibonacciSequence() ;
+        @Test
+        void Fibonacci_numbersFrom0to10() {
+            for (int index = 0; index < 10; index++) {
```
Comment (RubikOM) [inline src/test/java/com/varchenko/FibonacciSequenceTest.java]: You don't have to test several results in one test.
Tests are atomic. 
For one use case, we use 1 test.

By the way, what the difference between a test with value 6 and a value 8?

File: src/test/java/com/varchenko/FibonacciSequenceTest.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+            }
+        }
+
+        @Test
+        void Fibonacci_SpecificNumber() {
+            int expectedResult = fib(10);
+            int result = 55;
+
+            assertEquals(expectedResult, result);
+            System.out.println(fib(10));
```
Comment (RubikOM) [inline src/test/java/com/varchenko/FibonacciSequenceTest.java]: That's a good one. 

But please, don't use `System.out.println` in the tests if you not trying to print something useful. 
As a programmer, I won't take any information from `55` written to the console. 

P.S. you have the value in the `expectedResult ` field. Why do you call the method again? You just wasting compute time and RAM.

 P.P.S. You confused the `expectedResult ` and `result ` fields. Switch them back.

File: pom.xml [hunk 9]
```diff
@@ -5,7 +5,7 @@
     <modelVersion>4.0.0</modelVersion>
 
     <groupId>com.varchenko</groupId>
-    <artifactId>InsertSort</artifactId>
+    <artifactId>BubbleSort</artifactId>
```
Comment (RubikOM) [inline pom.xml]: This is not a `BubbleSort` project anymore. 
Pick some valid name. 

E.G. Sortings and algorithms.

File: src/test/java/com/varchenko/BubbleSortTest.java [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+class BubbleSortTest {
+    private final BubbleSort bubbleSort = new BubbleSort();
+
+    @Test
+    void sortArray_positiveNumbers() {
+        int[] arrayToSort = {5, 2, 1, 10, 50, 10};
+        int[] expectedResult = {1, 2, 5, 10, 10, 50};
+
+        int[] result = bubbleSort.sortArray(arrayToSort);
+System.out.println(Arrays.toString(arrayToSort));
```
Comment (RubikOM) [inline src/test/java/com/varchenko/BubbleSortTest.java]: Please remove this `System.out.println`.

Also, this line has been moved to the left. In IntelliJ, you can fix it via `Ctrl + Alt + O`.

File: src/test/java/com/varchenko/FibonacciSequenceTest.java (hunk 7 above)
Comment (RubikOM) [reply src/test/java/com/varchenko/FibonacciSequenceTest.java]: You had to add some edge case tests. 

For example, test what happens if requested 1, 0, -1, 5000.
Please add the following tests. 

Besides being a good practice, they will show you that your code is pretty undeveloped :)

**Review by RubikOM — CHANGES_REQUESTED** [review]

File: src/test/java/com/varchenko/FibonacciSequenceTest.java (hunk 7 above)
Comment (varchenkomichael) [reply src/test/java/com/varchenko/FibonacciSequenceTest.java]: done

File: src/test/java/com/varchenko/FibonacciSequenceTest.java (hunk 6 above)
Comment (varchenkomichael) [reply src/test/java/com/varchenko/FibonacciSequenceTest.java]: done

File: src/test/java/com/varchenko/FibonacciSequenceTest.java (hunk 8 above)
Comment (varchenkomichael) [reply src/test/java/com/varchenko/FibonacciSequenceTest.java]: done

File: pom.xml (hunk 9 above)
Comment (varchenkomichael) [reply pom.xml]: done

File: src/test/java/com/varchenko/BubbleSortTest.java (hunk 10 above)
Comment (varchenkomichael) [reply src/test/java/com/varchenko/BubbleSortTest.java]: done
